djstripe/models/base.py

```diff
@@ -573,12 +573,37 @@ def _stripe_object_to_invoice_items(cls, target_cls, data, invoice):
         :type invoice: ``djstripe.models.Invoice``
         """
 
-        lines = data.get("lines")
+        lines = data.get("lines", [])
         if not lines:
             return []
 
+        def iter_lines(lines):
+            """Return an iterator for all of the paginated lines from a Stripe list.
+
+            :param lines: Lines object belonging to a Stripe API object.
+            :type lines: dict
+            """
+
+            last_id = None
+
+            while True:
+                # yield each invoice item
+                for line in lines.get("data", []):
+                    last_id = line['id']  # update high water mark
+                    yield line
+
+                # stop iteration if all invoice items have been returned
+                if not lines.get("has_more", None):
+                    break
+
+                # fetch next page of invoice items from Stripe
+                lines = lines.list(
+                    limit=50,
+                    starting_after=last_id,
+                )
+
         invoiceitems = []
-        for line in lines.get("data", []):
```

Comment:
I think we can use ``auto_paging_iter`` provided by Stripe instead.

eg ``for line in lines.auto_paging_iter()):``

Reply:
oh, sweet, ty!

Reply:
Missed this feature in the Stripe docs -- I'll rewrite today